File: leetcode/graph/2737_Find_the_Closest_Marked_Node/codes/bellman_ford.py
import sys
from typing import List

class Solution:
    def minimumDistance(self, n: int, edges: List[List[int]], s: int, marked: List[int]) -> int:
        dist = [sys.maxsize] * n
        dist[s] = 0

        for _ in range(n - 1):
            for u, v, w in edges:
                if dist[u] != sys.maxsize and dist[u] + w < dist[v]:
                    dist[v] = dist[u] + w

        min_dist = min((dist[m] for m in marked), default=sys.maxsize)

        return -1 if min_dist == sys.maxsize else min_dist

File: leetcode/graph/2737_Find_the_Closest_Marked_Node/codes/bellman_ford.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minimumDistance(int n, vector<vector<int>>& edges, int s, vector<int>& marked) {
        vector<int> dist(n, INT_MAX);
        dist[s] = 0;

        for (int iter = 0; iter < n - 1; iter++) {
            for (auto& edge : edges) {
                int u = edge[0];
                int v = edge[1];
                int w = edge[2];
                if (dist[u] != INT_MAX && dist[u] + w < dist[v]) {
                    dist[v] = dist[u] + w;
                }
            }
        }

        int minDist = INT_MAX;
        for (int m : marked) {
            if (dist[m] < minDist) minDist = dist[m];
        }

        return minDist == INT_MAX ? -1 : minDist;
    }
};
